src/Xamarin.Android.Build.Tasks/Tasks/GenerateResourceCaseMap.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2021 Microsoft, Inc. All rights reserved.
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Android.Build.Tasks;
+
+namespace Xamarin.Android.Tasks
+{
+	public class GenerateResourceCaseMap : AndroidTask
+	{
+		public override string TaskPrefix => "GRCM";
+		public ITaskItem[] Resources { get; set; }
+
+		public string ResourceDirectory { get; set; }
+
+		public string ProjectDir { get; set; }
+
+		public ITaskItem[] AdditionalResourceDirectories { get; set; }
+
+		public ITaskItem OutputFile { get; set; }
+
+		private Dictionary<string, string> resource_fixup = new Dictionary<string, string> (StringComparer.OrdinalIgnoreCase);
+
+		public override bool RunTask ()
+		{
+			// ResourceDirectory may be a relative path, and
+			// we need to compare it to absolute paths
+			ResourceDirectory = Path.GetFullPath (ResourceDirectory);
+
+			// Create our capitalization maps so we can support mixed case resources
+			foreach (var item in Resources) {
+				var path = Path.GetFullPath (item.ItemSpec);
+				if (!path.StartsWith (ResourceDirectory, StringComparison.OrdinalIgnoreCase)) {
+					Log.LogDebugMessage ($"Skipping {item}. Path is not include the '{ResourceDirectory}'");
+					continue;
+				}
+
+				var name = path.Substring (ResourceDirectory.Length).TrimStart ('/', '\\');
```

Comment:
I'm not sure why we're "normalizing" `\` and `/` onto `/`, and then later in the `AddRename()` call we map from `/` back to `Path.DirectorySeparatorChar`.  Why is the normalization required?

Reply:
We are not normalising we are trimming the start of the path off. 

So if we have `obj\res\drawable\foo.png` and `ResourceDirectory` is `obj\res` we end up with `\drawable\foo.png` whch is not going to work with the lookup. So this is trimming off any leaving separators so the later lookups will work as expected. Because the `MonoAndroidResDirIntermediate` property which drives this is user editable we might have a path which is `obj\res\` or `obj\res`. We must have hit this before in the old code.